### PR TITLE
CIF-2577: add wish list to gift card products

### PR DESCRIPTION
**Pull request by Buuhuu**
## Description

In this PR the gift card product details got extended by an add to wish list button.

## Related Issue

CIF-2407

## How Has This Been Tested?

Locally and existing unit tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [X] All unit tests pass on CircleCi.
- [X] I ran all tests locally and they pass.

### CODE REVIEW

File: react-components/src/components/GiftCardOptions/giftCardOptions.js [hunk 1]
```diff
@@ -48,13 +49,21 @@ const messages = defineMessages({
...
+            changeOptionValue,
+            changeQuantity,
+            canAddToCart,
+            addToCart,
+            addToWishlist
+        }
     ] = useGiftCardOptions({ sku });
 
-    const { formatMessage } = useIntl();
+    const intl = useIntl();
```
Comment (herzog31) [inline react-components/src/components/GiftCardOptions/giftCardOptions.js:66]: Is this change necessary?

**Review by herzog31 — APPROVED** [review]

File: react-components/src/components/GiftCardOptions/giftCardOptions.js (hunk 1 above)
Comment (Buuhuu) [reply react-components/src/components/GiftCardOptions/giftCardOptions.js:66]: It is actually recommended to use `intl` that way to ease extraction. See the https://formatjs.io/docs/tooling/cli example

```
function Comp(props) {
  const {intl} = props
  return intl.formatMessage({
    // The whole `intl.formatMessage` is required so we can extract
    id: 'foo',
    defaultMessage: 'foo',
    description: 'bar',
  })
}
```

File: react-components/src/components/GiftCardOptions/useGiftCardOptions.js [hunk 2]
```diff
@@ -25,43 +25,41 @@ const useGiftCardOptions = props => {
...
-            if (!sku) {
-                console.error('SKU is not present in the dataset of mountpoint element');
-                return;
-            }
-            const { data, error } = await giftCardProductQuery({ variables: { sku }, fetchPolicy: 'network-only' });
-
-            if (error) {
-                throw new Error(error);
-            }
+    const fetchGiftCardOptions = useCallback(async () => {
```
Comment (laurentiumagureanu) [inline react-components/src/components/GiftCardOptions/useGiftCardOptions.js:28]: Why is this change necessary ?

Comment (Buuhuu) [reply react-components/src/components/GiftCardOptions/useGiftCardOptions.js:28]: This returns a function that is updated when the `sku` changes. And again when the function changes `useEffect` is called again to fetch the gift card options. Before the options were fetched only once. 

I used `useCallback` to create a memorised value as it is unlikely that the `sku` actually changes.

Comment (laurentiumagureanu) [reply react-components/src/components/GiftCardOptions/useGiftCardOptions.js:28]: why not use just the `useEffect` with a dependency on `sku` ?

Comment (Buuhuu) [reply react-components/src/components/GiftCardOptions/useGiftCardOptions.js]: done

**Review by laurentiumagureanu — APPROVED** [review]
